### DISCUSSION TITLE
Precision Date and Time (Local) help!

### DISCUSSION THREAD

**Discussion by JessHCarroll** [gopro/labs#428] [Q&A]
Hi, I used the Precision Date and Time (Local) labs (https://gopro.github.io/labs/control/precisiontime/) for research in march, and it's been fantastic to keep my GoPro time synced exactly (it's really important the time on the gopro is correct to the second). Unfortunately when I tried to use it today the seconds are constantly 3 seconds out from the correct time. Does anyone have any advice on how to fix this?

Alternatively I've tried the GPS time sync, but this always seems to be 1 second out?

Thanks so much!

Jess

**Comment by dnewman-gpsw** [comment]
What is your reference for correct time?  Precision Date and Time (Local) uses the time of the device used to display the QR Code. If that is wrong, then the camera will be wrong.  You clock source might need a reset, or resync with an external clock reference. There is no way, the camera is 3 second off from the time set in the QR Code.  Now there can be a processing delay, between reading the QR Code and setting the time, this is mostly calibrated out, but you can do your own calibrations using !MTCAL=(offset), so if time is off by -100ms, use  !MTCAL=-100  

GPS Sync should be good, although it can be wrong for new cameras for a while, as the transmission of GPS leap seconds is not very often.  Camera needs to see that signal once, normally a 10-minute of operation with clear GPS is sufficient to receive that signal, and that only needs to happen once.

**Comment by t0b3** [comment]
FTR: you may sync video dateTime from GPS after recording, i.e. using a shell script
```
#!/bin/bash
for filename in $* ; do
    [ -w $filename ] || return
    utcdate=$(exiftool -m -p /usr/share/exiftool/fmt_files/gpx.fmt -ee $filename | xq -r '.gpx.trk.trkseg.trkpt[-1].time')
    [ -n $utcdate ] || return
    localdate=$(date -d $utcdate '+%Y-%m-%dT%H:%M:%S')
    [ -n $localdate ] || return
    exiftool -overwrite_original -q -FileModifyDate="$localdate" -QuickTime:TrackCreateDate="$localdate" -QuickTime:TrackModifyDate="$localdate" -QuickTime:MediaCreateDate="$localdate" -QuickTime:MediaModifyDate="$localdate" -AllDates="$localdate" $filename
done
```
